### PR TITLE
hubble/relay: report nodes connectivity status on ServerStatus

### PR DESCRIPTION
**Pull request by rolinh**
Please, see individual commits for details.

### CODE REVIEW

File: pkg/hubble/relay/observer/server.go [hunk 1]
```diff
@@ -169,8 +172,15 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
 				"No connection to peer %s, skipping", p.Name,
 			)
 			s.peers.ReportOffline(p.Name)
+			// we dont want mean to be exhaustive here as reporting all
```
Comment (glibsm) [inline pkg/hubble/relay/observer/server.go]: if truncated, we should show something like `... and {{42}} more` a the end

File: pkg/hubble/relay/observer/server.go [hunk 2]
```diff
@@ -169,8 +172,15 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
...
+			// unavailable nodes might clutter in certain cases
+			// reporting up to 10 unavailable nodes is probably reasonnable
+			if len(unavailableNodes) < 10 {
```
Comment (glibsm) [inline pkg/hubble/relay/observer/server.go]: make a const up top

**Comment by glibsm** [comment]
side note: starting each commit message with `This commit...` is not needed. We know it's a commit, we're reading the commit message :) 

Instead of saying `This commit adds three new fields: a b c`, it's clearer to use `Add three new fields: a b c`

File: api/v1/Makefile [hunk 3]
```diff
@@ -51,4 +51,4 @@ clean:
 
 .PHONY: deps
 deps:
-	@$(PROTOC) --version | grep 3.11.4 || (echo "Error: protoc 3.11.4 is required" && exit 1)
```
Comment (glibsm) [inline api/v1/Makefile]: i wonder if there is a better way to vendor this in, so it's per-project, not system-wide

**Review by glibsm — APPROVED** [review]

File: pkg/hubble/relay/observer/server.go (hunk 1 above)
Comment (rolinh) [reply pkg/hubble/relay/observer/server.go]: Yes, I plan to do this when implementing the CLI counterpart of this PR.

**Comment by Rolinh** [comment]
> side note: starting each commit message with ...

Indeed, I've amended the commit messages and will keep this in mind for the upcoming ones.

File: api/v1/Makefile (hunk 3 above)
Comment (rolinh) [reply api/v1/Makefile]: I believe that @gandro wanted to update the Makefile to use a docker container instead of the host binary at some point.
I guess this would do.

**Comment by Rolinh** [comment]
test-me-please

File: api/v1/Makefile (hunk 3 above)
Comment (glibsm) [reply api/v1/Makefile]: Yeah that works for me. No action needed on this PR

File: api/v1/Makefile [hunk 4]
```diff
@@ -51,4 +51,4 @@ clean:
...
+	@$(PROTOC) --version | grep 3.12.3 || (echo "Error: protoc 3.12.3 is required" && exit 1)
```
Comment (aanm) [inline api/v1/Makefile]: This change does not seem relevant to the PR. I'm always in favor of updating things but this PR is also marked to be backported to 1.8. Ideally we should avoid YOLOing backporting vendor / generated changes into older branches.

**Review by aanm — CHANGES_REQUESTED** [review]

File: api/v1/Makefile (hunk 4 above)
Comment (rolinh) [reply api/v1/Makefile]: I agree with you in general but for this particular case I think it is fine as `protoc` is only used for to generate files. When using the newer version, the generated files are strictly identical.

Comment (rolinh) [reply api/v1/Makefile]: I removed this commit and opened #12677.

**Review by gandro — APPROVED** [review]

**Comment by Rolinh** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by Rolinh** [comment]
Marking as ready to merge as `generate-api` was just added and would need a rebase to pull the config. All tests have passed.
